lib/fetchers/TokenFetcher.ts

```diff
@@ -59,16 +59,20 @@ export class TokenFetcher {
   };
 
   public resolveToken = async (chainId: ChainId, address: string): Promise<Currency> => {
+    const token = await this.getTokenByAddress(chainId, address);
+    if (token === undefined) {
+      throw new ValidationError(`Could not find token with address ${address}`);
+    }
+    return token;
+  };
+
+  public getTokenByAddress = async (chainId: ChainId, address: string): Promise<Currency | undefined> => {
```

Comment:
Can't we use the above function `resolveTokenByAddress`?

Reply:
Oh nvm thats for addresses

Reply:
Weird that we have 4 different functions in this file for resolving tokens, each with slightly different logic / return values

Reply:
Okay I removed the functions we weren't really using / didn't need. Think this is more consolidated now.